File: selenium/test/java/com/thoughtworks/selenium/corebased/TestCheckUncheck.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.InternalSelenseTestNgBase;

import org.testng.annotations.Test;

public class TestCheckUncheck extends InternalSelenseTestNgBase {
	@Test(dataProvider = "system-properties") public void testCheckUncheck() throws Exception {
		selenium.open("../tests/html/test_check_uncheck.html");
		verifyTrue(selenium.isChecked("base-spud"));
		verifyFalse(selenium.isChecked("base-rice"));
		verifyTrue(selenium.isChecked("option-cheese"));
		verifyFalse(selenium.isChecked("option-onions"));
		selenium.check("base-rice");
		verifyFalse(selenium.isChecked("base-spud"));
		verifyTrue(selenium.isChecked("base-rice"));
		selenium.uncheck("option-cheese");
		verifyFalse(selenium.isChecked("option-cheese"));
		selenium.check("option-onions");
		verifyTrue(selenium.isChecked("option-onions"));
		verifyFalse(selenium.isChecked("option-chilli"));
		selenium.check("option chilli");
		verifyTrue(selenium.isChecked("option-chilli"));
		selenium.uncheck("option index=3");
		verifyFalse(selenium.isChecked("option-chilli"));
	}
}

File: dotnet/test/webdriverbackedselenium/TestCheckUncheck.cs
using NUnit.Framework;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestCheckUncheck : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldBeAbleToCheckAndUncheck()
        {
            selenium.Open("/html/test_check_uncheck.html");
            Assert.IsTrue(selenium.IsChecked("base-spud"));
            Assert.IsFalse(selenium.IsChecked("base-rice"));
            Assert.IsTrue(selenium.IsChecked("option-cheese"));
            Assert.IsFalse(selenium.IsChecked("option-onions"));
            selenium.Check("base-rice");
            Assert.IsFalse(selenium.IsChecked("base-spud"));
            Assert.IsTrue(selenium.IsChecked("base-rice"));
            selenium.Uncheck("option-cheese");
            Assert.IsFalse(selenium.IsChecked("option-cheese"));
            selenium.Check("option-onions");
            Assert.IsTrue(selenium.IsChecked("option-onions"));
            Assert.IsFalse(selenium.IsChecked("option-chilli"));
            selenium.Check("option chilli");
            Assert.IsTrue(selenium.IsChecked("option-chilli"));
            selenium.Uncheck("option index=3");
            Assert.IsFalse(selenium.IsChecked("option-chilli"));
        }
    }
}
